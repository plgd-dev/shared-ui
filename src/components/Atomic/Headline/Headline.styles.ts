import { css } from '@emotion/react'
import { HeadlineType } from './Headline.types'
import { colors } from '../_utils/colors'

const getSizes = (type: HeadlineType) => {
    switch (type) {
        case 'h1':
            return css`
                font-size: 72px;
                line-height: 80px;
                letter-spacing: -2px;

                @media (max-width: 767px) {
                    font-size: 50px;
                    line-height: 58px;
                }
            `
        case 'h2':
            return css`
                font-size: 48px;
                line-height: 54px;
                letter-spacing: -1.5px;

                @media (max-width: 767px) {
                    font-size: 36px;
                    line-height: 42px;
                }
            `
        case 'h3':
            return css`
                font-size: 36px;
                line-height: 44px;
                letter-spacing: -0.5px;

                @media (max-width: 767px) {
                    font-size: 24px;
                    line-height: 32px;
                }
            `
        case 'h4':
            return css`
                font-size: 24px;
                line-height: 30px;
                letter-spacing: -0.5px;

                @media (max-width: 767px) {
                    font-size: 20px;
                    line-height: 24px;
                }
            `
        case 'h5':
            return css`
                font-size: 20px;
                line-height: 24px;
                letter-spacing: -0.5px;

                @media (max-width: 767px) {
                    font-size: 18px;
                    line-height: 20px;
                }
            `
    }
}

export const headline = (type: HeadlineType) => css`
    color: #00c5c7 !important;
    font-family: 'Circular Pro', sans-serif;
    font-weight: 700;
    font-style: normal;
    ${getSizes(type)};
    margin: 0;
`
