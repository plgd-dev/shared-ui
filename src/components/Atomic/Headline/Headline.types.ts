import { ReactNode } from 'react'

export type HeadlineType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5'

export type Props = {
    className?: string
    children: ReactNode
    id?: string
    type?: HeadlineType
}

export const defaultProps: Partial<Props> = {
    type: 'h1',
}
