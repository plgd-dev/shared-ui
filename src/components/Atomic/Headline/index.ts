export { default } from './Headline'
export * from './Headline'
