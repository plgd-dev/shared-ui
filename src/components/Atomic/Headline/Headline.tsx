import { createElement, FC } from 'react'
import { Props, defaultProps } from './Headline.types'
import * as styles from './Headline.styles'
import classNames from 'classnames'

const Headline: FC<Props> = (props) => {
    const mergedProps = { ...defaultProps, ...props }
    const HeadlineElement = ({ type, children, ...props }: Props) => createElement(type as string, props, children)

    const { dataTestId, className, ...rest } = mergedProps

    return (
        <HeadlineElement
            {...rest}
            children={mergedProps.children}
            className={classNames(mergedProps.type, className)}
            css={styles.headline(mergedProps.type!)}
            data-test-id={mergedProps.dataTestId}
        />
    )
}

Headline.displayName = 'Headline'

export default Headline
