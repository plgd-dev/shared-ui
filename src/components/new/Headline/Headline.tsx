import { createElement, FC } from 'react'
import { Props, defaultProps } from './Headline.types'
import * as styles from './Headline.styles'
import classNames from 'classnames'

const Headline: FC<Props> = (props) => {
    const mergedProps = { ...defaultProps, ...props }
    const HeadlineElement = ({ type, children, ...props }: Props) => createElement(type as string, props, children)

    return (
        <HeadlineElement
            {...mergedProps}
            children={mergedProps.children}
            className={classNames(mergedProps.type, mergedProps.className)}
            css={styles.headline(mergedProps.type!)}
        />
    )
}

Headline.displayName = 'Headline'

export default Headline
