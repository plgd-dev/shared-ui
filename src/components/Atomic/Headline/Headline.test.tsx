import { render } from '../../../../test/jest-wrapper'
import Headline from './Headline'

describe('<Headline>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <Headline type='h1'>Headline 1</Headline>
                <Headline type='h2'>Headline 2</Headline>
                <Headline type='h3'>Headline 3</Headline>
                <Headline type='h4'>Headline 4</Headline>
                <Headline type='h5'>Headline 5</Headline>
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders correct text', () => {
        const { getByText } = render(<Headline type='h1'>Test</Headline>)
        expect(getByText('Test')).toBeInTheDocument()
    })

    it('applies correct class based on type', () => {
        const { container } = render(<Headline type='h1'>Test</Headline>)
        const headline = container.querySelector('h1')
        expect(headline).toHaveClass('h1')
    })

    it('applies h6 style when type is h6', () => {
        const { container } = render(<Headline type='h2'>Test</Headline>)
        const headline = container.querySelector('h2')
        expect(headline).toHaveClass('h2')
    })
})
