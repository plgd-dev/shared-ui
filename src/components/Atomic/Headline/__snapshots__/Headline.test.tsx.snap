// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: 'Circular Pro',sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 72px;
  line-height: 80px;
  letter-spacing: -2px;
  margin: 0;
}

@media (max-width: 767px) {
  .emotion-0 {
    font-size: 50px;
    line-height: 58px;
  }
}

.emotion-1 {
  font-family: 'Circular Pro',sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 48px;
  line-height: 54px;
  letter-spacing: -1.5px;
  margin: 0;
}

@media (max-width: 767px) {
  .emotion-1 {
    font-size: 36px;
    line-height: 42px;
  }
}

.emotion-2 {
  font-family: 'Circular Pro',sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 36px;
  line-height: 44px;
  letter-spacing: -0.5px;
  margin: 0;
}

@media (max-width: 767px) {
  .emotion-2 {
    font-size: 24px;
    line-height: 32px;
  }
}

.emotion-3 {
  font-family: 'Circular Pro',sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 24px;
  line-height: 30px;
  letter-spacing: -0.5px;
  margin: 0;
}

@media (max-width: 767px) {
  .emotion-3 {
    font-size: 20px;
    line-height: 24px;
  }
}

.emotion-4 {
  font-family: 'Circular Pro',sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 20px;
  line-height: 24px;
  letter-spacing: -0.5px;
  margin: 0;
}

@media (max-width: 767px) {
  .emotion-4 {
    font-size: 18px;
    line-height: 20px;
  }
}

<div>
    <h1
      class="h1 emotion-0"
    >
      Headline 1
    </h1>
    <h2
      class="h2 emotion-1"
    >
      Headline 2
    </h2>
    <h3
      class="h3 emotion-2"
    >
      Headline 3
    </h3>
    <h4
      class="h4 emotion-3"
    >
      Headline 4
    </h4>
    <h5
      class="h5 emotion-4"
    >
      Headline 5
    </h5>
  </div>
</DocumentFragment>
`;
